okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -1061,10 +1062,13 @@ open class OkHttpClient internal constructor(
     private fun newSslSocketFactory(trustManager: X509TrustManager): SSLSocketFactory {
       try {
         val sslContext = Platform.get().newSSLContext()
-        sslContext.init(null, arrayOf<TrustManager>(trustManager), null)
+        // no support for bouncycastle SecureRandom on Android
+        // see JcaTlsCryptoProvider.create
+        val random = SecureRandom()
```

Comment:
BouncyCastle provider doesn't have a working SecureRandom on Android, so this is required.

https://stackoverflow.com/questions/10259780/does-bouncycastle-have-a-securerandom-service

Reply:
@swankjesse Any risk with this change?  

https://github.com/bcgit/bc-java/blob/master/tls/src/main/java/org/bouncycastle/tls/crypto/impl/jcajce/JcaTlsCryptoProvider.java#L66

Looks like a reason to do this, but also a reason to only do this if you truly understand what it's doing.  We don't have that context as we don't control the JVM security config, our users do.